osu.Game/Screens/Play/Player.cs

```diff
@@ -236,11 +237,20 @@ public void Restart()
             });
         }
 
-        private void onPass()
+        private void onCompletion()
         {
             Delay(1000);
             Schedule(delegate
             {
+                // Force a final check to see if the player has failed
+                // Some game modes (e.g. taiko) fail at the end of the map
+                if (scoreProcessor.CheckFailed())

```

Comment:
Yeah you're right. It was there for a specific reason in the taiko branch, but the way this is being called now is much more sane. Will have to re-test with the taiko branch.